lib/rubocop/cop/style/empty_case_condition.rb

```diff
@@ -70,21 +70,24 @@ def correct_case_whens(corrector, case_node, when_nodes)
         end
 
         # Since an if condition containing commas is not syntactically valid, we
-        # correct `when x,y` to `if [x,y].any?`.
+        # correct `when x, y` to `if x || y`.
         def correct_multiple_alternative_whens(corrector, when_nodes)
           when_nodes.each do |when_node|
-            children = when_node.children
+            *children, _ = when_node.children
 
             # In `when a; r` we have two children: [a, r].
             # In `when a, b, c; r` we have 4.
-            next unless children.size > 2
+            # In `when a, b then r` we have 3.
+            next unless children.size > 1
 
+            first = children.first
+            last = children.last
             range =
               Parser::Source::Range.new(when_node.loc.expression.source_buffer,
-                                        children[0].loc.expression.begin_pos,
-                                        children[-2].loc.expression.end_pos)
+                                        first.loc.expression.begin_pos,
+                                        last.loc.expression.end_pos)
 
-            corrector.replace(range, children[0..-2].map(&:source).join(' || '))
+            corrector.replace(range, children[0..-1].map(&:source).join(' || '))
```

Comment:
I think the children need to be parenthesised between `||`s, here's a contrived example:
```
case
when 1 < 2 ? nil : false, 1
  puts 'ok'
else
  puts 'urk'
end
```
which is corrected to:
```
if 1 < 2 ? nil : false || 1
  puts 'ok'
else
  puts 'urk'
end
```

but that has a different behaviour when evaluated (since `||` is higher precedence than `? :`):
```
irb(main):001:0> case
irb(main):002:1* when 1 < 2 ? nil : false, 1
irb(main):003:1>   puts 'ok'
irb(main):004:1> else
irb(main):005:1*   puts 'urk'
irb(main):006:1> end
ok
=> nil
irb(main):007:0> if 1 < 2 ? nil : false || 1
irb(main):008:1>   puts 'ok'
irb(main):009:1> else
irb(main):010:1*   puts 'urk'
irb(main):011:1> end
urk
=> nil
```